src/ssl.c

```diff
@@ -24759,49 +24775,198 @@ WOLFSSL_X509_LOOKUP_METHOD* wolfSSL_X509_LOOKUP_file(void)
     return &meth;
 }
 
+/* set directory path to load certificate or CRL which have the hash.N form */
+/* for late use                                                             */
+/* @param ctx    a pointer to WOLFSSL_BY_DIR structure                      */
+/* @param argc   directory path                                             */
+/* @param argl   file type, either WOLFSSL_FILETYPE_PEM or                  */
+/*                                          WOLFSSL_FILETYPE_ASN1           */
+/* @return WOLFSSL_SUCCESS on successful, othewise negative or zero         */
+static int x509AddCertDir(void *p, const char *argc, long argl)
```

Comment:
Is there a reason why "p" is a void* here instead of a WOLFSSL_BY_DIR*?  I only see x509AddCertDir() called once in ssl.c and it passes in a WOLFSSL_BY_DIR.

Reply:
I cannot remember why I use void* here. Might be I put it as void* when moving function into ssl.c/internal.c,and calling it from several source files.
Anyway, there is no reason of "void*". Changed to WOLFSSL_BY_DIR*. Also, added sanity check for ctx pointer.